service/Checkout.php

```diff
@@ -24,10 +24,56 @@
 
 namespace Adyen\PrestaShop\service;
 
+use Adyen\PrestaShop\service\adapter\classes\ServiceLocator;
+use Cart;
+
 class Checkout extends \Adyen\Service\Checkout
 {
+    const PAYMENT_METHOD_STEP = 'checkout-payment-step';
+    const DELIVERY_STEP = 'checkout-delivery-step';
+    const IS_REACHABLE = 'step_is_reachable';
+    const IS_COMPLETE = 'step_is_complete';
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
     public function __construct(Client $client)
     {
         parent::__construct($client);
+        $this->logger = ServiceLocator::get('Adyen\PrestaShop\service\Logger');
+    }
+
+    /**
+     * Check if the payment method step is next in the checkout process
+     *
+     * @param Cart $cart
+     *
+     * @return bool
+     */
+    public function isPaymentMethodStepNext(Cart $cart)
+    {
+        try {
+            $checkoutSessionData = \Db::getInstance()->getValue(
+                'SELECT checkout_session_data FROM ' . _DB_PREFIX_ . 'cart WHERE id_cart = ' . (int)$cart->id
+            );
+
+            $jsonData = json_decode($checkoutSessionData, true);
```

Comment:
please add some defensive programming here

if the checkoutSessionData is empty then we can early return with the error
if the json_decode failed as well (try catch won't catch it) so you can do something similar to
```
if ($jsonData === null && json_last_error() !== JSON_ERROR_NONE) {
   echo "json data is incorrect";
}
```

Reply:
https://github.com/Adyen/adyen-prestashop/pull/207/commits/7a886af752f3521ac358b05509f8f9d2f2a9c41d